pkg/identityserver/store/end_device.go

```diff
@@ -40,6 +40,7 @@ type EndDevice struct {
 	ModelID         string `gorm:"type:VARCHAR"`
 	HardwareVersion string `gorm:"type:VARCHAR"`
 	FirmwareVersion string `gorm:"type:VARCHAR"`
+	BandID          string `gorm:"type:VARCHAR"`
```

Comment:
In `EndDevice`, this now competing with `FrequencyPlanID`. That field is not stored in IS, but in NS. I do think it is valuable to keep this here. @rvolosatovs thoughts?

Reply:
This is still open

Reply:
Since FrequencyPlanID maps to single a BandID, but a single BandID can be used by multiple Frequency Plans, I do not know what the action should be here. Note that this BandID is only used to fetch the correct profile from the device repository.

Maybe @rvolosatovs can provide some more context?

Reply:
Sorry, but I lack context myself to say anything as it's unclear to me why band ID would be required to fetch a device profile. Also I don't know what IS has to do with this.
If you could explain what is happening here and why I could give you my thoughts.

In any case, device profile message in the backend interfaces contains `RFRegion`, which corresponds to our frequency plan ID perfectly fine I don't see why we'd ever want to use just the band ID here, since we can always derive band ID from the frequency plan ID.

Reply:
Also see https://github.com/TheThingsNetwork/lorawan-devices/blob/master/vendor/the-things-products/the-things-uno.yaml#L9-L17. In short, for the Device Repository there is a device profile per region.

Reply:
@rvolosatovs 

>In any case, device profile message in the backend interfaces contains `RFRegion`, which corresponds to our frequency plan ID perfectly fine I don't see why we'd ever want to use just the band ID here, since we can always derive band ID from the frequency plan ID.

It corresponds to the Band ID, that's the issue here.

So my question is, is the `BandID` in the IS authoritive or informative? If the former, should NS respect this value, and validate frequency plan IDs, or should we keep this decoupled? Decoupling may get inconsistent devices: band EU863-870 with frequency plan US902-915 FSB 2.

Please consider this from the developer perspective when dealing with our APIs.

Reply:
This `BandID` is used (by the AS) to retrieve the appropriate end device codecs from the Device Repository. Two things are of note here: First, AS is not aware of the Frequency Plan ID used by the device (and so it cannot infer the Band ID). Secondly, the frequency plan ID for the device is chosen as a separate step.

Indeed, validating the BandID with respect to the Frequency Plan ID would make sense when creating the device. However, the Network Server is also unaware of the version ids.

What should be the course of action here?

Reply:
I think it's not the cleanest approach, but it makes more sense to validate the frequency plan in NS, especially from developer/customer standpoint.

Reply:
Added a relevant check in `pkg/networkserver/grpc_deviceregistry.go`, but I see that https://github.com/TheThingsNetwork/lorawan-stack/pull/3680 is changing everything there. I think it is best to revert (to avoid merge conflicts) add a TODO for this.